lib/server/ServerRenderer.js

```diff
@@ -59,14 +63,29 @@ function injectScriptAtBottomOfBody(html, script) {
 }
 
 function makeClientAppStartScript(clientAppRequirePath, environmentConfig, bootstrappedData) {
-    return "<script type=\"text/javascript\">\n" +
-        "var ClientApp = require('" + clientAppRequirePath + "');\n" +
+    return "var ClientApp = require('" + clientAppRequirePath + "');\n" +
         "var clientApp = new ClientApp();\n" +
         "clientApp.start({\n" +
         "environmentConfig: " + environmentConfig + ",\n" +
         "bootstrappedData: " + bootstrappedData + "\n" +
-        "});\n" +
-        "</script>";
+    "});";
+}
+
+function makeBootstrappingLoader(scriptBody, clientAppUrl, isAsync) {
+    return "<script type=\"text/javascript\">\n" +
+        "\"use strict\";" +
+        "(function(global) {\n" +
+        "var s = document.createElement(\"script\");\n" +
+        "s.setAttribute(\"src\", \"" + clientAppUrl + "\");\n" +
+        "s.setAttribute(\"async\", " + !!isAsync + ");\n" +
+        "function loaded(e) {\n" +
+            scriptBody + "\n" +
+            "s.removeEventListener(\"load\", loaded, false);\n" +
+        "}\n" +
+        "s.addEventListener(\"load\", loaded);\n" +
+        "global.document.body.appendChild(s);\n" +
+    "})(window);\n" +
```

Comment:
Can also inject document here so as to not have to do window.document inside the script.